includes/form-fields.php

```diff
@@ -0,0 +1,98 @@
+<?php
+    // initiate possible user roles from settings
+    $possible_ignore_roles = false;
+    
+    if(is_admin()){
+        global $wp_roles;
+        $possible_ignore_roles = array();
```

Comment:
You populate the `$possible_ignore_roles` only if `is_admin()` returns `true`. After that you check if there are ignore roles in order to initialize the `$this->form_fields['ignore_for_roles']`. You can do all of that here, in this `if`:

```php
if(is_admin()){
        global $wp_roles;
        $possible_ignore_roles = array();
        
        foreach($wp_roles->roles as $role => $stuff){
            $possible_ignore_roles[$role] = $stuff['name'];
        }

        $this->form_fields['ignore_for_roles'] = array(
            'title'             => __( 'Ignore tracking for roles', 'metrilo-analytics' ),
            'type'              => 'multiselect',
            'description'       => __( '<strong style="color: #999;">(Optional)</strong> If you check any of the roles, tracking data will be ignored for WP users with this role', 'metrilo-analytics' ),
            'desc_tip'          => false,
            'default'           => '',
            'options'           => $possible_ignore_roles
        );
    }
```

Thus you are not going to expose the `$possible_ignore_roles` outside of the scope where it is being used and you won't need the extra `if` starting at line 30.